client-core/src/service/key_service.rs

```diff
@@ -21,32 +21,32 @@ where
         KeyService { storage }
     }
 
-    /// Adds a new public-private keypair to storage
-    pub fn add_keypair(
+    /// Adds a new wallet_name-private keypair to storage
+    pub fn add_wallet_private_key(
         &self,
+        wallet_name: &str,
         private_key: &PrivateKey,
-        public_key: &PublicKey,
         enckey: &SecKey,
     ) -> Result<()> {
         self.storage
             .set_secure(
                 KEYSPACE,
-                public_key.serialize(),
+                wallet_name.as_bytes(),
```

Comment:
why public-key is not necessary?

Reply:
because the public-key is the view key, already stored in `Wallet`